dramatiq/worker.py

```diff
@@ -332,11 +360,12 @@ def handle_message(self, message):
             message.fail()
             self.post_process_message(message)
 
-    def post_process_message(self, message):
+    def post_process_message(self, message: MessageProxy):
         """Called by worker threads whenever they're done processing
         individual messages, signaling that each message is ready to
         be acked or rejected.
         """
+        assert self.consumer
```

Comment:
Guard this with an `if TYPE_CHECKING` to show that this assertion is for the benefit of static type checking only?

Reply:
It still understands `asserts` and uses them for [type narrowing](https://mypy.readthedocs.io/en/stable/type_narrowing.html#type-narrowing-expressions). Looks like 
```python
if TYPE_CHECKING:
    assert self.consumer 
```
works just fine and at least makes the intent of the `assert` clearer to the reader. What do you think?

Reply:
Ok good point about the type narrowing. This is good then.